marklogic-data-hub/src/main/java/com/marklogic/hub/dhs/DhsDeployer.java

```diff
@@ -48,13 +51,27 @@ public void deployAsDeveloper(HubConfigImpl hubConfig) {
     }
 
     public void deployAsSecurityAdmin(HubConfigImpl hubConfig) {
+        throwExceptionIfMarkLogicVersionIsInvalid(hubConfig);
         prepareAppConfigForDeployingToDhs(hubConfig);
 
         HubAppDeployer dhsDeployer = new HubAppDeployer(hubConfig.getManageClient(), hubConfig.getAdminManager(), null, null);
         dhsDeployer.setCommands(buildCommandsForSecurityAdmin());
         dhsDeployer.deploy(hubConfig.getAppConfig());
     }
 
+    public void throwExceptionIfMarkLogicVersionIsInvalid(HubConfig hubConfig) {
+        MarkLogicVersion mlVersion = new MarkLogicVersion(hubConfig.getManageClient()
```

Comment:
I think we need a friendly try/catch on these too, in case this call fails - which could happen for a number of reasons, most often that the user's creds are incorrect. 

Also, I think we can put that try/catch and the code in the MarkLogicVersion class. It should have a constructor for accepting a String version, which makes testing it easy. But it should also have a constructor that takes a ManageClient. It will then make this call itself, wrapping it in a try/catch. The message can be "Unable to get version of MarkLogic; cause: " + ex.getMessage()

That will avoid the duplication of getting the ML version too, which is also being done in DeployHubAmpsCommand.

Reply:
Added the try/catch and moved the code to MarkLogicVersion class.